modules/ref_fb_module/tests/test_fb_statistics.cpp

```diff
@@ -426,3 +518,42 @@ TEST_F(StatisticsTest, StatisticsTestChangingOutputDomain)
                                        newOutputDomain);
     helper.run();
 }
+
+TEST_F(StatisticsTest, StatisticsTestTriggerBasic)
+{
+    vecvec<Float> mockPackets{{0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0},
+                              {1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2.0},
+                              {2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 3.0},
+                              {0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0},
+                              {1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2.0},
+                              {2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 3.0}};
+    vecvec<Float> expectedAvg{{}, {}, {2.35}, {0.95}, {}, {}};
+    vecvec<Float> expectedRms{{}, {}, {2.3674881203503428}, {1.3946325680981353}, {}, {}};
+    vecvec<Int> expectedDomain{{}, {}, {41}, {61}, {}, {}};
+    Int initialOutputDomain = 0;  // Implicit
+
+    std::vector<Int> triggerModeChangesBeforePackets{0};                     // TODO test multiple changes
+    std::vector<Bool> newTriggerMode{true};                                  // TODO test true true, false false
+    vecvec<Float> mockTriggerPackets{{0.1, 0.6, 0.1}, {}, {}, {}, {}, {}};   // TODO test sending while TriggerMode = false
+    vecvec<Int> mockTriggerDomainPackets{{26, 40, 90}, {}, {}, {}, {}, {}};  // TODO test sending trigger packets on time or later
+
+    auto helper = StatisticsTestHelper(LinearDataRule(2, 5),
+                                       initialOutputDomain,
+                                       expectedAvg,
+                                       expectedRms,
+                                       expectedDomain,
+                                       SampleTypeFromType<Float>::SampleType,
+                                       mockPackets,
+                                       {},
+                                       {},
+                                       {},
+                                       {},
+                                       {},
+                                       triggerModeChangesBeforePackets,
+                                       newTriggerMode,
+                                       mockTriggerPackets,
+                                       mockTriggerDomainPackets);
+    helper.run();
+}
+
+// TODO add more trigger as nested function block tests, test multiple trigger condition changes in same packet
```

Comment:
I don't think too many more tests are needed, mostly I'd add a test where the trigger goes from 0 to 1 several times.